lib/charon/config.ex

```diff
@@ -58,7 +58,12 @@ defmodule Charon.Config do
     # 15 minutes
     access_token_ttl: 15 * 60,
     enforce_browser_cookies: false,
-    json_module: Jason,
+    json_module:
+      if :erlang.system_info(:otp_release) |> to_string() |> String.to_integer() >= 27 do
```

Comment:
I think this fails when running Elixir 1.17 on Erlang/OTP 27, [which is supported](https://hexdocs.pm/elixir/1.17.1/changelog.html#erlang-otp-support).

Erlang/OTP added `:json` in [version 27](https://www.erlang.org/blog/highlights-otp-27/#the-new-json-module), but Elixir only exposed the `JSON` module in [1.18](https://github.com/elixir-lang/elixir/releases/tag/v1.18.0). I would rather check against Elixir's minor version:

`System.version() |> String.split(".") |> Enum.at(1) |> String.to_integer() >= 18`

Reply:
hm good point, I got confused because they added native json support to erlang originally, but of course the elixir version has to actually have the module too :man_facepalming:

Reply:
Elixir's `JSON` implementation doesn't actually call Erlang's! It is fully independent, albeit partially copied from the OTP implementation: https://github.com/elixir-lang/elixir/blob/main/lib/elixir/src/elixir_json.erl.